src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -236,27 +237,27 @@ private MessagePayloadDataConverter CreateErrorDataConverter(IErrorSerializerSet
             return new MessagePayloadDataConverter(errorSerializerSettingsFactory.CreateJsonSerializerSettings(), isDefault);
         }
 
-        private IDurabilityProviderFactory GetDurabilityProviderFactory(ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
+        private IDurabilityProviderFactory GetDurabilityProviderFactory(DurableTaskOptions options, ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
         {
-            bool storageTypeIsConfigured = this.Options.StorageProvider.TryGetValue("type", out object storageType);
+            bool storageTypeIsConfigured = options.StorageProvider.TryGetValue("type", out object storageType);
+
             if (!storageTypeIsConfigured)
             {
                 logger.LogInformation("Using the default storage provider: Azure Storage.");
-                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(DefaultProvider));

```

Comment:
I recognize that this should never happen, but I wonder if we should try to fail here if `DefaultProvider` isn't found. Perhaps we can change `FirstOrDefault()` to just `First()`? An exception thrown here might be easier to debug that a null-reference exception thrown by the caller.

Reply:
That makes sense. I changed both `FirstOrDefault()` calls in this method to `First()` to keep it consistent.